viewer/api/views.py

```diff
@@ -77,3 +77,83 @@ def metric(request, metric_id):
 
     serializer = DistributionSerializer(qs, populations=pops, subgroup=sg)
     return Response(serializer.data)
+
+
+# API v2
+
+@api_view(['GET'])
+def experiments(request):
+    datasets = (
+        DataSet.objects.visible()
+                       .order_by(F('created_at').desc(nulls_last=True))
+    )
+    data = []
+    for d in datasets:
+        data.append({
+            'id': d.id,
+            'name': d.name,
+            'completed': True,  # TODO
+        })
+
+    return Response({'experiments': data})
+
+
+@api_view(['GET'])
+def experiment_by_id(request, exp_id):
+    try:
+        dataset = DataSet.objects.visible().get(id=exp_id)
+    except DataSet.DoesNotExist:
+        raise NotFound('No experiment with given id found.')
+
+    data = {
+        'id': dataset.id,
+        'name': dataset.name,
+        'description': '',  # TODO
+        'authors': [],
+        'populations': dataset.get_populations(),
+        'subgroups': dataset.get_subgroups(),
+        'metrics': dataset.get_metrics(),
+    }
+    return Response(data)
+
+
+@api_view(['GET'])
+def metric_by_id(request, exp_id, metric_id):
+    metric = Metric.objects.get(id=metric_id)
+    # TODO: Add subgroups.
+    pops = Collection.objects.filter(dataset=exp_id,
+                                     metric=metric_id,
+                                     subgroup='All').order_by('population')
+
+    populations = []
+    for p in pops:
+        if metric.type == 'UintScalar':
+            populations.append({
+                'name': p.population,
+                'n': p.num_observations,
+                'data': [
+                    dict(x=d.bucket,
```

Comment:
Not a huge deal, but would it be possible to return the correct type here? I notice in [1030/300](http://localhost:8000/v2/experiments/1030/metrics/300/), for example, that all of the x-values are numbers but they're represented as strings.

Reply:
We moved to making them strings so the charting lib didn't try to interpret them as numbers and adjust scales. I think all the data so far matches the histogram type of chart where the x-value isn't really a point but a range (e.g. `[0-1)` but is represented as x=0).

Reply:
I'm not sure if I understand what you mean by the x value being a *range* rather than a point. Either way, I do convert the strings to ints before it goes to the charting library, anyway, so I wouldn't worry about that.

Reply:
I don't know what the right answer is here at the moment. I'm storing them as strings on the backend so it's not clear what the "correct type" from the point of view of this code. I think in the future we may have some x-values that are not integers?

I'd prefer to keep it similar to what EV is doing and we can open an issue to discuss it in more detail?

Reply:
Alright, works for me. Let's stick with strings for now and if it becomes an issue we'll deal with it then.